src/test/java/uk/nhs/hee/tis/trainee/sync/migration/CopyProgrammeMembershipToCurriculumMembershipTest.java

```diff
@@ -0,0 +1,136 @@
+  private static final String DEST_CLASS_NAME = CurriculumMembership.class.getName();
+
+  private CopyProgrammeMembershipToCurriculumMembership migration;
+
+  private MongoTemplate template;
+  private DefaultIndexOperations indexOperationsSource;
+  private DefaultIndexOperations indexOperationsDest;
+  private MongoCollection<Document> sourceCollection;
+  private MongoCollection<Document> destCollection;
+
+  @BeforeEach
+  void setUp() {
+    template = mock(MongoTemplate.class);
+    indexOperationsSource = mock(DefaultIndexOperations.class);
+    indexOperationsDest = mock(DefaultIndexOperations.class);
+    sourceCollection = mock(MongoCollection.class);
+    destCollection = mock(MongoCollection.class);
+    migration = new CopyProgrammeMembershipToCurriculumMembership(template);
+  }
+
+  @Test
+  void shouldCreateCurriculumMembershipTable() {
+    IndexField idxFieldSimple = IndexField.create("simpleIndexField", Sort.Direction.ASC);
+    IndexField idxFieldCompound1 = IndexField.create("compoundIndexField1", Sort.Direction.ASC);
+    IndexField idxFieldCompound2 = IndexField.create("compoundIndexField2", Sort.Direction.ASC);
+    IndexInfo idxInfoSimple = new IndexInfo(List.of(idxFieldSimple), "test", false, false, "en");
+    IndexInfo idxInfoCompound = new IndexInfo(List.of(idxFieldCompound1, idxFieldCompound2), "test2", false, false, "en");
+
+    //given
+    when(template.collectionExists(DEST_COLLECTION)).thenReturn(false);
+    when(template.indexOps(SOURCE_COLLECTION)).thenReturn(indexOperationsSource);
+    when(template.indexOps(DEST_COLLECTION)).thenReturn(indexOperationsDest);
+    when(indexOperationsSource.getIndexInfo()).thenReturn(List.of(idxInfoSimple, idxInfoCompound));
+
+    //when
+    migration.createCurriculumMembershipTable();
+    ArgumentCaptor<Index> indexCaptor = ArgumentCaptor.forClass(Index.class);
+
+    //then
+    verify(template).createCollection(DEST_COLLECTION);
+    verify(indexOperationsDest, times(2)).ensureIndex(indexCaptor.capture());
+    List<Index> indexList = indexCaptor.getAllValues();
+    assertThat("Unexpected simple index field sort order",
+        indexList.get(0).getIndexKeys().get("simpleIndexField"),
+        is(1));
+    assertThat("Unexpected compound index field sort order",
+        indexList.get(1).getIndexKeys().get("compoundIndexField2"),
+        is(1));
+  }
+
+  @Test
+  void shouldCopyProgrammeMembershipTable() {
+    Document doc = new Document();
+    doc.put("_id", "1");
+    doc.put("_class", "old.class");
+
+    FindIterable iterable = mock(FindIterable.class);
+    MongoCursor cursor = mock(MongoCursor.class);
+
+    //given
+    when(template.getCollection(SOURCE_COLLECTION)).thenReturn(sourceCollection);
+    when(template.getCollection(DEST_COLLECTION)).thenReturn(destCollection);
+
+    when(sourceCollection.find()).thenReturn(iterable);
+    doCallRealMethod().when(iterable).forEach(any(Consumer.class)); //to allow the iterator to catch forEach()
```

Comment:
This feels a **_lot_** like the test is knowing too much about the inner workings of the method under test, but I'm not sure how else to establish reasonable coverage at this point.